### PR TITLE
[GPU] Fix colorization-sig accuracy issue using oneDNN

### PR DESCRIPTION
**Pull request by hyunback**
Memory crash in case reuse_eltwise_sum_post in oneDNN and memory_pool
And print node in/out gpu_usm_mem addr at OV_GPU_Verbose >= 1

Signed-off-by: hyunback <hyunback.kim@intel.com>

### Tickets:
 - *80380*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/network.cpp [hunk 1]
```diff
@@ -700,6 +700,15 @@ void network::execute_impl(const std::vector<event::ptr>& events) {
...
                            << inst->output_memory().get_allocation_type() << std::endl;
+            auto debug_in_out_mem_addr = [&]() {
+                for (size_t i = 0; i < get_primitive(inst->id())->dependencies().size(); i++) {
+                    auto& in_mem = get_primitive(inst->id())->dep_memory(i);
+                    GPU_DEBUG_COUT << "In " << in_mem.buffer_ptr() << std::endl;
+                }
+                auto& out_mem = get_primitive(inst->id())->output_memory();
+                GPU_DEBUG_COUT << "Out " << out_mem.buffer_ptr() << std::endl;
+            };
+            debug_in_out_mem_addr();
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/network.cpp:711]: why did you define a lambda here? Could you print those buf addr in a single line?

File: src/plugins/intel_gpu/src/graph/graph_optimizer/basic_memory_dependencies.cpp [hunk 2]
```diff
@@ -36,14 +36,13 @@ void basic_memory_dependencies::run(program& p) {
...
         }
 
-        if (node->is_type<convolution>() && node->get_preferred_impl_type() == impl_types::onednn) {
-            auto& conv = node->as<convolution>();
+        if (node->get_preferred_impl_type() == impl_types::onednn) {
             bool can_reuse_eltwise_mem = false;
             size_t eltw_dep = 0;
 
-            for (auto& fused_op : conv.get_fused_primitives()) {
+            for (auto& fused_op : node->get_fused_primitives()) {
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/basic_memory_dependencies.cpp:43]: This is too general. For example, `pooling` does not support sum post operation.

File: src/plugins/intel_gpu/src/graph/network.cpp (hunk 1 above)
Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/network.cpp]: updated it

**Comment by isanghao** [comment]
Performance looks good. Let's merge it and follow-up for remaining things. @jade-cho

**Pull request closed without merging**